examples/data_preprocessing/[WIP]_chexpert_dataset/Prepare_CheXpert_data.ipynb

```diff
@@ -0,0 +1,958 @@
+    "\n",
+    "In the following, a transform sequence is defined, which will be used to preprocess the images. The images are first resized to 224 x 224 pixels, since this is a requirement for many pre-trained CNNs in PyTorch, and then normalized with the mean and standard deviation from ImageNet. Such transformations have previously been applied by several other submissions regarding CheXpert on GitHub such as:\n",
+    "[this](https://github.com/gaetandi/cheXpert/blob/master/cheXpert_final.ipynb) and [this](https://github.com/Stomper10/CheXpert/blob/master/CheXpert_DenseNet121_FL.ipynb)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "transform_list = transforms.Compose([\n",
+    "    transforms.Resize((224, 224)),\n",
+    "    transforms.ToTensor(),\n",
+    "    transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225]) # normalization from ImageNet\n",
+    "    ])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The following class will be used to transform the images and prepare the dataset for PyTorch's DataLoader. Moreover, the uncertainty labels will be handled according to either the \"U-Ones\", \"U-Zeroes\", \"U-Ignore\" or \"U-MultiClass\" approach as they were described in the origianl CheXpert paper by Irvin et al. (2019).\n",
+    "\n",
+    "You can specify how the uncertainty labels for each individual pathology will be handled by providing a list of the names of the pathologies to the arguments to_ones, to_zeros and to_ignore. to_ignore changes the uncertainty label (-1.0) to nan. Not specifying any pathology in any of the arguments to_ones, to_zeros or to_ignore will train the model with the \"U-MultiClass\" approach. \n",
+    "\n",
+    "The \"Support Device\" label as well as the \"No Finding\" label is dropped for all instances, since support devices do not count as pathologies and the \"No Finding\" label only depends on the labels for the remaining 12 pathologies. The \"blank\" labels, indicating that an observation was not mentioned in the corresponding report, will be treated as \"uncertain\".\n",
+    "\n",
+    "If the return_image parameter is set to False, only the labels are returned. This might be useful if you already have the transformed images saved but wish to experiment with different settings of to_ones, to_zeros or to_ignore.\n",
+    "\n",
+    "The validation set can be loaded by setting the subset input of the class to \"valid\" and supplying the appropriate image paths and number of images.\n",
+    "\n",
+    "The structure of the class was inspired by the GitHub submissions credited above."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class CheXpertDatasetProcessor(Dataset):\n",
```

Comment:
why does it inherit from the `Dataset` class?

Reply:
I mainly chose that because most tutorials I looked up used it together with DataLoader (e.g.: https://pytorch.org/tutorials/beginner/basics/data_tutorial.html). Would something else be more suitable?

Reply:
Usually, if you inherit from some class, the "child" class shouldn't change the semantics of the "parent" class (e.g. if your parent class is Dataset (=the class that describes special datatype), you child class is not supposed to be the one where you *read* and *process* data). 

Also, the inheritance is usually needed when you want to reuse some arguments and/or functions from the parent class. Since you are not using any torch dataset properties here,  I don't think you need any inheritance here at all.

Reply:
Thanks, that's good to know! I assumed that adding Dataset was necessary for DataLoader to work (since I mainly saw the 2 together) , but you're right, it seems to work without it, so I'll remove it.